source/sigv4.c

```diff
@@ -1702,14 +1713,24 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
                                                   &noOfHeaders,
                                                   canonicalRequest );
 
-        if( ( sigV4Status == SigV4Success ) && !FLAG_IS_SET( flags, SIGV4_HTTP_HEADERS_ARE_CANONICAL_FLAG ) )
+        if( sigV4Status == SigV4Success )
         {
-            /* Sorting headers based on keys. */
-            quickSort( canonicalRequest->pHeadersLoc, noOfHeaders, sizeof( SigV4KeyValuePair_t ), cmpHeaderField );
+            if( FLAG_IS_SET( flags, SIGV4_HTTP_HEADERS_ARE_CANONICAL_FLAG ) )
+            {
+                /* Headers are already canonicalized, so just write it to the buffer as is. */
+                sigV4Status = writeLineToCanonicalRequest( pHeaders,
```

Comment:
I thought if the headers are already canonicalized then we don't need to write them to the buffer right here? @gshvang, is that still the case?

Reply:
Looks like this line was just moved here from elsewhere